tests/Integration/PrestaShopBundle/Controller/Sell/Customer/Address/AddressControllerTest.php

```diff
@@ -0,0 +1,249 @@
+    {
+        $router = $client->getKernel()->getContainer()->get('router');
+        $createAddressUrl = $router->generate('admin_addresses_create');
+        $crawler = $client->request('GET', $createAddressUrl);
+        $submitButton = $crawler->selectButton('save-button');
+        $addressForm = $submitButton->form();
+        $addressForm = $this->formFiller->fillForm($addressForm, $this->getAddressModifications());
+        $client->submit($addressForm);
+        $dataChecker = $client->getContainer()->get('test.integration.core.form.identifiable_object.data_handler.address_form_data_handler_checker');
+        $this->testAddressId = $dataChecker->getLastCreatedId();
+    }
+
+    /**
+     * Gets modifications that are needed to fill address form
+     *
+     * @return array
+     */
+    public function getAddressModifications(): array
+    {
+        $testAddress = $this->getTestAddress();
+
+        return [
+            'customer_address[customer_email]' => 'pub@prestashop.com',
+            'customer_address[alias]' => 'test_alias',
+            'customer_address[first_name]' => $testAddress->getFirstName(),
+            'customer_address[last_name]' => $testAddress->getLastName(),
+            'customer_address[address1]' => $testAddress->getAddress(),
+            'customer_address[postcode]' => $testAddress->getPostCode(),
+            'customer_address[city]' => $testAddress->getCity(),
+            'customer_address[id_country]' => 21,
+        ];
+    }
+
+    /**
+     * @param Crawler $crawler
+     * @param array $formModifications
+     *
+     * @return Form
+     */
+    private function fillFiltersForm(Crawler $crawler, array $formModifications): Form
+    {
+        $button = $crawler->selectButton('address[actions][search]');
+        $filtersForm = $button->form();
+        $this->formFiller->fillForm($filtersForm, $formModifications);
+
+        return $filtersForm;
+    }
+
+    /**
+     * @param Crawler $crawler
+     *
+     * @return Address[]
+     */
+    private function getAddressList(Crawler $crawler): array
+    {
+        return $crawler->filter('#address_grid_table')->filter('tbody tr')->each(function ($tr, $i) {
+            return $this->getAddress($tr, $i);
+        });
+    }
+
+    /**
+     * Default addresses are not fit for filtering very well because most of the information is identical. So I need
+     * new address with unique values to test various filters
+     *
+     * @return Address
+     */
+    private function getTestAddress(): Address
+    {
+        return new Address(
+                $this->testAddressId,
+                'testfirstname',
+                'testlastname',
+                'testaddress1',
+                '11111',
+                'testcity',
+                'lithuania'
+        );
+    }
+
+    /**
+     * I think it makes sense to return an DTO here instead of array. It doesn't make sense to reuse EditableCustomerAddress
```

Comment:
Using a DTO : great idea!

However as you mentioned, we need a better name to avoid confusion.

I would suggest to be very practical: AddressListDTOUsedForTesting, AddressListDTOForTest or TestAddressDTO...

Reply:
Yeah name change makes sense. But where do you think it belongs from architecture standpoint? It's meant strictly for tests so I guess it should i stay in tests folder?

Reply:
If it's meant strictly for tests it must stay in tests folder ✅ 
If it's meant to be used only by a class it should be as close as possible to this class ✅ 

So current location is the best IMO